### DISCUSSION TITLE
skrypt lista domen do bind9

### DISCUSSION THREAD

**Discussion by ham3r78** [MajkiIT/polish-ads-filter#21378] [Ideas]
Witam, plik hostfile.txt jest listą domen. 
Czy jest jakiś skrypt, którym przerobie to na plik, gdzie linijka po linijce będzie miała taką formę;
zone "blokowanadomena.pl" { type master; notify no; file "/etc/bind/blocked.zone"; };
wcześniej korzystalem z:
wget -O - https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt | grep '^127.0.0.1' | tr "[A-Z]" "[a-z]" | awk '!a[$0]++' | awk '{print "zone \""$2"\" { type master; notify no; file \"/etc/bind/blocked.zone\"; };"}' > /etc/bind/named.conf.blocked
ale tam każda linijka zaczyna sie od 127.0.0.1
a może jest gdzieś publikowana aktualna wersja właśnie w takiej formie?
(list domen używam do prytwatnego dns blokującego reklamy)
pozdrawiam.

**Comment by xxcriticxx** [comment]
@ham3r78 would it be better to install something like pihole or adguard?

**Reply by ham3r78** [reply]
yes, but it could working on server inside and outside my home network - connecting PCs, mobiles etc., as a extra feature to working server

**Reply by xxcriticxx** [reply]
pihole or adguard both work on the outside of the network if they configure correctly.

**Reply by ham3r78** [reply]
ok, i'll try :D thanks

**Reply by xxcriticxx** [reply]
@ham3r78 you can try both in docker

**Comment by krystian3w** [comment]
8 lat temu było: https://github.com/RemeJuan/Ubuntu-DNS-Adblocker/blob/master/blacklist-adaway

Niby 1Hosts wspiera Bind9 (/Kont / PowerDNS) ale w innej składni: https://github.com/badmojr/1Hosts
